spyder/widgets/mixins.py

```diff
@@ -848,6 +848,47 @@ def get_word_at(self, coordinates):
 
         return word
 
+    def get_uri_at(self, coordinates):

```

Comment:
Why did you add these methods to the mixins? You have to be aware that these mixins are shared by the Internal console and the IPython one. And since you're not adding URI detection to those, I think these methods should be moved to a better place (perhaps to CodeEditor itself, or at least TextEditBaseWidget).

Reply:
By these methods I mean both `get_uri_at` and `get_pattern_cursor_at`.